cpp/src_prims/utils.h

```diff
@@ -230,6 +230,15 @@ IntType gcd(IntType a, IntType b) {
   return a;
 }
 
+/** gets maximum available shared mem for the current device */
+inline size_t maxSharedMem() {
+  int devId;
+  CUDA_CHECK(cudaGetDevice(&devId));
+  cudaDeviceProp prop;
+  CUDA_CHECK(cudaGetDeviceProperties(&prop, devId));
```

Comment:
Isn't it possible to use `cudaDeviceGetAttribute()` instead?

Reply:
Ah.. this function is not used anywhere. We use `getSharedMemPerBlock` instead. Let me remove this one.